### PR TITLE
Pull Flink image from ECR

### PR DESCRIPTION
**Pull request by thodson-usgs**
This PR would pull the Flink image from the ECR public registry. This should avoid the ImagePullBackOff Error we were hitting for larger clusters (though I haven't tested at scale yet).

It also fixes a couple issues from the history server PR, like deleting the "default" security group and adding an optional permission boundary to the efs role.

### CODE REVIEW

File: terraform/aws/flink-operator-helm-values.yaml.tftpl [hunk 1]
```diff
@@ -0,0 +1,30 @@
+defaultConfiguration:
+  flink-conf.yaml: |+
+    #### defaultConfiguration from pangeo-forge-cloud-federation ####
+    # on aws pull base image from ecr
+    kubernetes.container.image.ref: ${flink_image_registry}flink:${flink_version}
```
Comment (ranchodeluxe) [inline terraform/aws/flink-operator-helm-values.yaml.tftpl:5]: nice find, didn't know about this. I wonder if this means we can leave this completely out of the runner or now if the runner overrides this value with deploy-specific precedence?

https://github.com/pangeo-forge/pangeo-forge-runner/blob/main/pangeo_forge_runner/bakery/flink.py#L247C9-L254

**Review by ranchodeluxe — APPROVED** [review]
looks good, non-blocking question

File: terraform/aws/flink-operator-helm-values.yaml.tftpl (hunk 1 above)
Comment (thodson-usgs) [reply terraform/aws/flink-operator-helm-values.yaml.tftpl:5]: Ouch, my spinning head. Fine to stew on this while Flink is on the shelf.

Comment (thodson-usgs) [reply terraform/aws/flink-operator-helm-values.yaml.tftpl:5]: Ah, but we do need to merge those other fixes to enable anything post `history-server` on USGS's cloud account.

Comment (ranchodeluxe) [reply terraform/aws/flink-operator-helm-values.yaml.tftpl:5]: this should be fine b/c the deployment and `image:<>` tags override what is here but those won't be pointing to the ECR version. let's find out

**Pull request merged**